fix: missing dependencies in quick mode temporary project

## Bug

`barodoc serve` and `barodoc build` fail in quick mode (zero-config) with:

```
Cannot find module '@astrojs/mdx' imported from '@barodoc/theme-docs/src/index.ts'
```

## Cause

`packages/barodoc/src/runtime/project.ts` generates a temporary `.barodoc/package.json` that only lists `astro`, `@barodoc/core`, `@barodoc/theme-docs`, `react`, and `react-dom`.

`@barodoc/theme-docs` imports `@astrojs/mdx`, `@astrojs/react`, `@tailwindcss/vite`, `@tailwindcss/typography`, and `tailwindcss`, but these are in its `devDependencies` (not `dependencies`), so `npm install` in the temp project doesn't install them.

## Fix

Add the missing packages to the generated `package.json` dependencies in `project.ts`:
- `@astrojs/mdx`
- `@astrojs/react`
- `@tailwindcss/typography`
- `@tailwindcss/vite`
- `tailwindcss`

--- a/.changeset/fix-quick-mode-deps.md
+++ b/.changeset/fix-quick-mode-deps.md
@@ -1,0 +1,7 @@
+---
+"barodoc": patch
+---
+
+Fix missing dependencies in quick mode temporary project
+
+Add @astrojs/mdx, @astrojs/react, @tailwindcss/vite, @tailwindcss/typography, and tailwindcss to the generated .barodoc/package.json so that `barodoc serve` and `barodoc build` work in zero-config mode.

--- a/packages/barodoc/src/runtime/project.ts
+++ b/packages/barodoc/src/runtime/project.ts
@@ -101,6 +101,11 @@ export async function createProject(options: ProjectOptions): Promise<string> {
       private: true,
       dependencies: {
         astro: "^5.0.0",
+        "@astrojs/mdx": "^4.0.0",
+        "@astrojs/react": "^4.0.0",
+        "@tailwindcss/typography": "^0.5.19",
+        "@tailwindcss/vite": "^4.0.0",
+        tailwindcss: "^4.0.0",
         "@barodoc/core": "^1.0.0",
         "@barodoc/theme-docs": "^1.0.0",
         react: "^19.0.0",

